spark/src/main/scala/io/substrait/spark/expression/ToSparkExpression.scala

```diff
@@ -44,6 +44,15 @@ class ToSparkExpression(
       Literal.FalseLiteral
     }
   }
+
+  override def visit(expr: SExpression.I8Literal): Expression = {
+    Literal(expr.value().asInstanceOf[Byte], ToSubstraitType.convert(expr.getType))
+  }
+
+  override def visit(expr: SExpression.I16Literal): Expression = {
+    Literal(expr.value().asInstanceOf[Short], ToSubstraitType.convert(expr.getType))
```

Comment:
and here also the other direction for the conversion?

Reply:
It's already covered in the other direction in `ToSubstraitExpression` on the following line:
`case SubstraitLiteral(substraitLiteral) => Some(substraitLiteral)`
The `unapply` method in this object invokes the conversion.